daml-lf/spec/transaction.rst

```diff
@@ -282,6 +284,12 @@ In this version, these fields are included:
 
 ``maintainers`` must be non-empty.
 
+*since version 9*
+
+The key may not contain contract IDs. This change is backwards
+incompatible; that is, transactions containing contract IDs in keys
+cannot be deserialized any more.
```

Comment:
As mentioned in a previous comment

> it would be improper to only mention this restriction in a since version 9 section in transaction.rst (as commented elsewhere), since you are imposing the restriction retroactively.

That's because you can always read the spec such that if you are dealing with version _n_, you can ignore all sections _since version e_ where _e > n_; that is not the case here.

Effectively, the restriction you are introducing is _since version 3_, in other words, on all `KeyWithMaintainers`.